proposals/20210126-nc-and-ncp.md

```diff
@@ -0,0 +1,218 @@
+
+### Non-Goals/Future Work
+
+Non-goals are limited to the scope of this document, these features will evolve
+over time.
+
+- To support managing real nodes in CAPN. _Technically, CAPN can use real nodes. But we have to incorporate an entire new machine provision mechanism/workflow which we leave as the future work._
+- To support managing components running in the super cluster worker nodes. _All the node plugins such as Kubeproxy, CNI/CSI will be managed in the super cluster by the super cluster administrator. They cannot be configured through the CAPN APIs._
+- To support a separate scheduler in each CAPN control plane. _There are few use cases which require two level schedulers but it is out of the scope for CAPN as of now._ 
+- Define how each [component controller](/proposals/00_capn-glossary.md#component-controller) operates, this is defined in [Creating Control Plane Components](/proposals/20201026-creating-control-plane-components.md)
+
+
+## Proposal
+
+### User Stories
+
+1. As a control plane operator, I want to be able to set the namespace where my cluster is provisioned
+2. As a control plane operator, I want to be able to set the name of my control plane
+3. As a control plane operator, I want to be able to specify the [component controllers](/proposals/00_capn-glossary.md#component-controller) for my cluster.
+4. As a machine, I want to know how to address my control plane.
+
+### Features from user stories
+
+#### NestedCluster
+
+Kubernetes API Group: `infrastructure.cluster.x-k8s.io/v1alpha4`
+
+This resource is responsible for getting the Cluster API `Cluster` type and setting the `spec.infrasteructureRef` pointing back to to this `NestedCluster`, this allows the CAPI controllers to understand the state of the cluster. As well It listens to the `NestedControlPlane` resource which is associated and keeps it's status in sync as the cluster is provided.
+
+
+```go=
+type NestedClusterSpec struct {
+	// ControlPlaneEndpoint represents the endpoint used to communicate with the control plane.
+	// +optional
+	ControlPlaneEndpoint clusterv1.APIEndpoint `json:"controlPlaneEndpoint"`
+}
+
+type NestedClusterStatus struct {
+	// Ready is when the NestedControlPlane has a API server URL.
+	// +optional
+	Ready bool `json:"ready,omitempty"`
+}
+
+type NestedCluster struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	Spec   NestedClusterSpec   `json:"spec,omitempty"`
+	Status NestedClusterStatus `json:"status,omitempty"`
+}
+```
+
+#### NestedControlPlane
+
+Kubernetes API Group: `controlplane.cluster.x-k8s.io/v1alpha4`
+
+The NestedControlPlane resource is responsible for orchestrating the overarching cluster, this controller doesn't create any of the downstream objects instead it creates a place where the downstream objects can look up shared values. 
+
+**Question:**
+1. should we have information necessary for the syncer within this object? 
```

Comment:
I think the only thing the syncer needs is the `kubeconfig` of the NestedControlPlane. We used to store the kubeconfig in a secret on the super cluster.